### ISSUE TITLE
Error when calling findFirst even though nothing has changed

### ISSUE THREAD

**Issue by camchenry** [mswjs/data#78]
First, thanks for the great library. This solves the exact problem I have been working on, and we're already using MSW so this is a natural fit.

I have been running into an issue where the relations seem to "disappear" from the database, even though nothing has changed? I created a CodeSandbox which consistently crashes:
https://codesandbox.io/s/mswjs-relation-undefined-error-wkcfp?file=/src/App.tsx
```tsx
import { factory, oneOf, primaryKey } from "@mswjs/data";
import * as faker from "faker";

const db = factory({
  user: {
    id: primaryKey(faker.datatype.uuid),
    firstName: String
  },
  userObject: {
    id: primaryKey(faker.datatype.uuid),
    data: String,
    user: oneOf("user")
  }
});

// Seed the database
const seededUser = db.user.create({
  id: "ab4f631c-cca4-498f-a5aa-4828352a7c69",
  firstName: "Test"
});
db.userObject.create({
  user: seededUser,
  data: "test data 1 - associated with user"
});

const queryObject = () => {
  const object = db.userObject.findFirst({
    where: {
      user: {
        id: {
          equals: "ab4f631c-cca4-498f-a5aa-4828352a7c69"
        }
      }
    }
  });
  console.log("query:", { object, user: object.user });
};

// This will work OK, user is not undefined
queryObject();

// This will NOT work ok
setTimeout(() => {
  console.log("deferred query");
  queryObject();
}, 1000);
```

I make the same `queryObject` call twice: once immediately after creation, and another 1 second after seeding the database. My expectation is that both invocations of the function should do exactly the same thing. However, what happens is that the first call works as expected, while the second invocation crashes with the error `TypeError: can't access property "__type", actualValue is undefined`.

Willing to help out with this, let me know what questions you have. Thanks in advance.

(I apologize for all of the issues, but I am desperately trying to figure out this issue, and I have had no luck. I'd love to use this project.)

**Comment by marcosvega91** [comment]
Hi @camchenry thanks for raising the issue :).

I think that making the property [`enumerable`](https://github.com/mswjs/data/blob/f8da857b3ea62dc2486b13f7cf9228e15e173b0d/src/model/defineRelationalProperties.ts#L82) should solve the issue. I have already done it in the same PR as for the other issue because I need it for deleting internal properties.

**Comment by camchenry** [comment]
@marcosvega91 Sounds great! I will look forward to when that releases then.

**Comment by kettanaito** [comment]
Thanks for raising this, @camchenry. You've been immensely helpful in uncovering these behaviors. That's indeed an issue and we should fix it in the upcoming release.

@marcosvega91, I'll take a look at the internal properties removal pull request once more, would love to get it merged. It's a superb piece of work already, so huge thanks! I have one concern, but we'll figure it out.

**Comment by kettanaito** [comment]
The issue itself is addressed in #67, but we need to add a test for it. @camchenry, please, would you be interested in contributing the relevant test? We can support you throughout the process and have this not only fixed, but bullet-proof against any future regressions?

**Comment by camchenry** [comment]
@kettanaito Sure! I'll try to get around to it today and put up a PR.

**Comment by kettanaito** [comment]
With the test added, is there anything left to address to mak this as done, @camchenry?

**Comment by camchenry** [comment]
@kettanaito i think with the tests written and the manual testing I've done, I think the original issue should be fixed 🤞

**Comment by camchenry** [comment]
@kettanaito Any chance this will make it into a new release sometime soon? This was somewhat of a blocker in my project, but I'd like to give it another go and see if I can uncover any more issues.

**Comment by kettanaito** [comment]
Released in [0.3.0](https://github.com/mswjs/data/releases/tag/v0.3.0). Please give it a try, your feedback is highly valuable!
